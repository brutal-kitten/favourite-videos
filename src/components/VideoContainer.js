import React, { Component } from 'react';
import Infobox from './Infobox';
import Picture from './Picture';
import ButtonsPanel from './ButtonsPanel';
import ShowVideo from './ShowVideo';
import Modal from './Modal'

class VideoContainer extends Component {

  state = {
    title: '',
    date: '',
    likes: '',
    views: '',
    thumbnails: '',
    error: false,
    showModal: false
  }

  playVideo = (id) => {
    this.setState({showModal: true});
  }

  closeModal = () => {
    this.setState({showModal: false})
  }

  componentDidMount() {
    if(this.props.fetchAgain) {
      const url = `https://www.googleapis.com/youtube/v3/videos?id=${this.props.id}&key=AIzaSyB7asSzTvcMogycBslu8o4RB3DjOumaqtA&part=snippet,contentDetails,statistics,status`
      fetch(url)
        .then((response) => response.json())
        .then((result) => {
          console.log(result.items[0]);
          if(result.items[0]) {
            this.setState({
              fetchAgain: false,
              title: result.items[0].snippet.title,
              date: result.items[0].snippet.publishedAt,
              views: result.items[0].statistics.viewCount,
              likes: result.items[0].statistics.likeCount,
              thumbnails: result.items[0].snippet.thumbnails.default.url
            });
          } else {
            this.props.changeSearchResultError(true);
            this.props.deleteVideo(this.props.id);

          }

        })
      }
  }

  render() {



    return (
        <div className="grid-container" id="videocontainer">
          {this.state.error === true && (
            <div className="noresult">
            <h3>There is no such video</h3>
            </div>
          )}
          <Picture
            thumbnails={this.state.thumbnails}
            playVideo={this.playVideo}
            id={this.props.id}
          />
          <ButtonsPanel
            id={this.props.id}
            isfavorite={this.props.isfavorite}
            addToFavorite={this.props.addToFavorite}
            deleteVideo={this.props.deleteVideo}
            playVideo={this.playVideo}
            removeFromFavorite={this.props.removeFromFavorite}
          />
          <Infobox
            title={this.state.title}
            date={this.state.date}
            views={this.state.views}
            likes={this.state.likes}
          />
          <Modal
            videoId={this.props.id}
            showModal={this.state.showModal}
            closeModal={this.closeModal}
          />
        </div>
    )
  }
};

export default VideoContainer;
