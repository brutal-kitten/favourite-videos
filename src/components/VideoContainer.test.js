import React from 'react';
import ReactDOM from 'react-dom';
import VideoContainer from './VideoContainer.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <VideoContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});
const mockPlayVideo = jest.fn();
const mockDeleVideo = jest.fn();
const mockAddToFavorite = jest.fn();
const mockRemoveFromFavorite = jest.fn();

const props = {
  title: "name",
  date: new Date("2017-05-31T09:30:02.000Z").toDateString(),
  likes: '8359285',
  views: '349487',
  id: '4759q3',
  playVideo: mockPlayVideo,
  deleteVideo: mockDeleVideo,
  isfavorite: true,
  addToFavorite: mockAddToFavorite,
  removeFromFavorite: mockRemoveFromFavorite,
  thumbnails: 'some url'
}
const wrapper = mount(<VideoContainer {...props}/>);
const initialState = {
  showModal: false
}

it('initialize VideoContainer with correct initial state', () => {
  expect(wrapper.state()).toEqual(initialState);
});


describe('should have given given props', () => {

  it('should have given title', () =>  {
    expect(wrapper.prop('title')).toBe(props.title);
  });

  it('should have given date', () =>  {
    expect(wrapper.prop('date')).toBe(props.date);
  });

  it('should have given likes', () =>  {
    expect(wrapper.prop('likes')).toBe(props.likes);
  });

  it('should have given views', () =>  {
    expect(wrapper.prop('views')).toBe(props.views);
  });
  it('should have given id', () =>  {
    expect(wrapper.prop('id')).toBe(props.id);
  });

  it('should have given playVideo ', () =>  {
    expect(wrapper.prop('playVideo')).toBe(props.playVideo);
  });

  it('should have given deleteVideo ', () =>  {
    expect(wrapper.prop('deleteVideo')).toBe(props.deleteVideo);
  });

  it('should have given isfavorite', () =>  {
    expect(wrapper.prop('isfavorite')).toBe(props.isfavorite);
  });

  it('should have given addToFavorite ', () =>  {
    expect(wrapper.prop('addToFavorite')).toBe(props.addToFavorite);
  });

  it('should have given image url', () =>  {
    expect(wrapper.prop('thumbnails')).toBe(props.thumbnails);
  });
});


describe("state can be changed by playVideo and showModal", () => {

  it('method closeModal should set state to false', () => {

    const wrapper2 = shallow(<VideoContainer {...props}/>);
    wrapper2.setState({
      showModal: true
    });
    wrapper2.instance().closeModal();
    expect(wrapper2.state('showModal')).toBe(false);
  });

  it('method playVideo should set state to true', () => {

    const wrapper3 = shallow(<VideoContainer {...props}/>);
    wrapper3.instance().playVideo();
    expect(wrapper3.state('showModal')).toBe(true);
  });

});
