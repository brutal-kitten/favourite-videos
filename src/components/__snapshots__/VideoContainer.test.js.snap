// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="grid-container videocontainer"
>
  <div
    className="pic grid-item"
  >
    <img
      alt="thumbnail"
      onClick={[Function]}
      tabIndex="0"
    />
  </div>
  <div
    className="panel grid-item"
  >
    <div
      className="play"
    >
      <button
        className="buttonPlay"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="glyphicon glyphicon-play-circle"
        />
      </button>
    </div>
    <div
      className="delete"
    >
      <button
        className="buttonDelete"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="glyphicon glyphicon-trash"
        />
      </button>
    </div>
  </div>
  <div
    className="info"
  >
    <div
      className="title infobox"
    />
    <div
      className="addedAt infobox"
    />
    <div
      className="likes infobox"
    >
      <span />
       
      <span
        className="glyphicon glyphicon-heart"
      />
    </div>
    <div
      className="views infobox"
    >
      <span />
       views
    </div>
  </div>
</div>
`;
